consumer/consumer.go

```diff
@@ -115,16 +136,22 @@ func (c *Consumer) handlePartitions(ch <-chan cluster.PartitionConsumer) error {
 			c.wg.Add(1)
 			go func(pc cluster.PartitionConsumer) {
 				defer c.wg.Done()
-
-				c.handleMessages(pc.Messages(), c.consumer, pc)
+				c.handleMessages(pc.Messages(), c.consumer, pc, pc.Topic(), pc.Partition())
 			}(part)
 		case <-c.quit:
+			if c.Logger != nil {
```

Comment:
:tulip: We don't need to check if the Logger is nil I guess, the `setup` function is there to make sure the pointer is not nil at startup. If after that at some point someone changes the pointer, it will panic and its fine. We should add a comment above the `Logger` field to indicate that this must not be changed after the `Serve` method. Something like `Logger blablabla. Must be set before Serve().`

Reply:
ok.